### ISSUE TITLE
Push configuration from existing environment file

### ISSUE THREAD

**Issue by andyshinn** [deis/deis#1532]
it would be nice to have a command to push configuration from an existing `.env` file. There already appears to be a `config:pull` command to download the `.env` file. This is a little bit related to #1531 in that it would help configure existing Procfile based applications in one go.

**Comment by gvilarino** [comment]
+1000

I think the `CLI` api should mimic [`ddolar/heroku-config`](https://github.com/ddollar/heroku-config) (a plugin for `heroku toolbelt`)

`deis config:pull` should keep it's functionality
`deis config:push` should push contents of a `.env` (if present) as if it were `deis config:set [..]`

**Comment by bacongobbler** [comment]
`deis config:pull` is already implemented. Personally I'm interested why a `config:push` command is necessary when `deis config:set $(cat .env)` does the same thing... Any reasons why the latter won't work?

**Comment by gvilarino** [comment]
tl;dr: I think most users would expect `config:push` to work if `config:pull` works.

I wouldn't call it 'necessary' as much as I would call it 'expected'. It's pretty safe to assume that over 80% of Deis implementors have been or still are Heroku users. With that in mind, the most sensible approach would be to mimic Heroku's API as much as possible, in order to provide what's expected by Deis' userbase. That being said, Heroku dosen't provide `config:pull` out of the box; that feature is instead provided by the `[heroku-config](https://github.com/ddollar/heroku-config)` plugin. But that plugin provides `config:push` as well, all based on the `.env` contents.

**Comment by bacongobbler** [comment]
I understand the motivation behind convenience functions, but after giving this more thought on this I believe that they should be a plugin, much like Heroku's client. If there's a way we can make the client more pluggable, this functionality could easily be added into the client and convenience functions like `config:pull` and `config:push` could be moved out into a plugin. I'm all for it; I'd just prefer that we focus on the "core" functionality and allow users to extend the client the way they prefer.

Does that seem reasonable to you?

**Comment by gvilarino** [comment]
:+1: 
I agree on `config:push/pull` being plugins as a right approach. I just thought that as `pull` was part of the core, then `push` might be just as well :)

**Comment by carmstrong** [comment]
> I just thought that as pull was part of the core, then push might be just as well

That's a valid point. @bacongobbler we probably need to have a larger discussion around plugin architecture.
